### ISSUE TITLE
Is there a reproducible dreambooth training recipe for this blog?

### ISSUE THREAD

**Issue by Victor49152** [huggingface/diffusers#2465]
I see the examples in [blog](https://huggingface.co/blog/dreambooth) generating very solid examples of dreambooth. But with the dog [example](https://drive.google.com/drive/folders/1BO_dyz-p65qhBRRMRA4TbZ8qW4rB99JZ), I tried to train for 800 steps but after 200 steps, whatever prompts I use gives me something like the original photo, without any background and context. I tried to train with/without prior preservation, but context is easily lost in both cases.

It would be very helpful if there is any reproducible example like in the blog can be provided. I find it very difficult to determine if this dreambooth finetuning is working as expected.

Following is my results which does not seem very good:

200-steps: sks dog in a bucket 
![image](https://user-images.githubusercontent.com/111467530/220796153-544cfcb8-0f2c-444b-aa0d-8592e3e4fc39.png)
400-steps: sks dog in a bucket
![image](https://user-images.githubusercontent.com/111467530/220796219-aa0e4ed8-d23d-4f0e-91b4-390ea8b2ca44.png)

My training command

accelerate launch train_dreambooth.py \
  --pretrained_model_name_or_path=runwayml/stable-diffusion-v1-5 \
  --train_text_encoder \
  --instance_data_dir=/datasets/instance_dir \
  --output_dir=/workspace/diffusers/output_model \
  --instance_prompt="a photo of sks dog" \
  --resolution=512 \
  --train_batch_size=1 \
  --class_prompt="a photo of dog" \
  --num_class_images=200 \
  --gradient_accumulation_steps=1 \
  --learning_rate=2e-6 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --class_data_dir=/datasets/dog \
  --max_train_steps=1600 \
  --checkpointing_steps 200

**Comment by sayakpaul** [comment]
Ccing @patil-suraj @pcuenca who are the authors of that blog post. 

Have you tried the following already?

* Different schedulers with different amounts of inference steps.

**Comment by Victor49152** [comment]
Hi @sayakpaul, thanks for cc this message. I have tried to use DDIM with 100 inference steps as suggested in the blog post, which is shown above. Not sure what else combinations make sense.

I might have not expressed myself clearly. The image quality is one thing, another thing is why I use batch size 1 and after 200 steps the context is already lost. In the blog post, I think they mentioned training with batch size 4 with ~400 steps. Since I don't have a reproducible example, I was wondering if I'm running the code correctly. Thank you so much for your response!!

**Comment by sayakpaul** [comment]
I think if you're changing the batch size then maybe adjust your learning rate accordingly. 200 training steps might be too low for this.

**Comment by e4s2022** [comment]
Hi @Victor49152 

I also found the similar observation as you described above. The learning rate was fixed as 2e-6, and the batch size is fixed as 1.  I tried following different hyper-parameters tunings:

(1)Fine-tuning the model for 400 steps
![Dreambooth-dog-lr2e6-step400](https://user-images.githubusercontent.com/107909552/221095710-172800b6-579f-4010-817b-85f2c9837280.png) The reconstruction looks good to me, but the bicycle I would like to see is not generated.

(2)Fine-tuning the model for 400 steps, with prior preservation loss enabled(300 dog images are provided)
![Dreambooth-dog-lr2e6-step400-wPrior](https://user-images.githubusercontent.com/107909552/221095741-58b5faff-02cb-4e56-add6-ed537fc164f4.png) The bicycle concept is captured, but recon. failed.

(3)Fine-tuning the model for 400 steps, with prior preservation loss enabled(300 dog images are provided), with textEncoder trainable
![Dreambooth-dog-lr2e6-step400-wPrior-trainTextEncoder](https://user-images.githubusercontent.com/107909552/221095774-cf688b47-2662-4831-9173-c152de38c804.png) The results are similar as in (1), but I found the overall quality is much better when fine-tuning the text encoder along with the UNet. But...unluckily...bicycle is still not found. And I get a pure black image occasionally.

During inference, I used the DDIM scheduler `(num_inference_steps=50, guidance_scale=7.5)` as suggested in the blog, and the prompt I gave is "a photo of sks dog riding a bicycle". The produced images seem to loss the context information no matter what prompt I gave.

**Comment by Victor49152** [comment]
Hi @e4s2022, 
Yes, this agrees with my experiments.
In summary

1. Only tuning unet loses context easily
2. With prior preservation loss, reconstruction of the injected object tends to fail when you give a different prompt than the instance prompt, but if use identical intance prompt, probably still fine with reconstruction.
3. With prior preservation loss and tuning text encoder, context is still easily to be lost, (I found around 200 steps we are still able to see some context.)

I also see black images, if you look into your log, there is something like they believe there is inappropriate content in this image so a black image is returned.

BTW, I see in this https://huggingface.co/spaces/multimodalart/dreambooth-training site, the help message mentioned that this training could fix the text encoder after ~30% of total steps. I don't know if that gonna help on above issues, maybe worth trying, but not mentioned in the blog or readme though.

**Comment by sayakpaul** [comment]
The black images are there because of the `safety_checker`.

Thanks,
Sayak | sayak.dev

**Comment by e4s2022** [comment]
@Victor49152 I noticed the original Dreambooth paper works on the Imagen, don't know if this is the reason. But I think the SD should also work in theory, at least from the results of that blog.

@sayakpaul Oh, I see.. Thx
